docs/src/tutorials/write-my-software-management-plugin.md

```diff
@@ -0,0 +1,393 @@
+# Write my software management plugin
+
+**thin-edge.io** provides Software Management plugins natively for APT (Debian).
+However, there are many package management systems in the world,
+and you may want to have a plugin that is suitable for your device.
+For such a demand, we provide **Software Management Plugin API**
+to write a custom Software Management plugin in your preferred language.
+
+In this tutorial, we will look into the **Software Management Plugin API**,
+and learn how to write your own plugin with a docker plugin shell script example.
+
+## Create a plugin
+
+Create a _docker_ file in the directory _/etc/tedge/sm-plugins/_. 
+A plugin must be an executable file and located in the directory.
+
+Filename: /etc/tedge/sm-plugins/docker
+
+```shell
+#!/bin/bash
```

Comment:
It is safer to use shell instead of bash as it may not be available on every machine.
```suggestion
#!/bin/sh
```

Reply:
The first docker example can run by sh, however, the `update-list` example runs only by bash. I copied the script from Christoph's example. Should I rewrite it to sh?

Reply:
I don't think there is a need to rewrite but maybe just annotate it such that it is clear.